lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -127,7 +127,12 @@ def discover
 
       new_backends = []
       @zk.children(@discovery['path'], :watch => true).each do |id|
-        node = @zk.get("#{@discovery['path']}/#{id}")
+        begin
+          node = @zk.get("#{@discovery['path']}/#{id}")
+        rescue ZK::Exceptions::NoNode => e
+          log.error("synapse: consistency issue reading #{id} at #{@discovery['path']}: #{e}")
```

Comment:
i guess the idea here is that we found the node in the list, but then it disappeared before we could read it's contents. so, we will omit it from `new_backends`. should we make the log message more explanatory? people are going to be confused by what a `consistency issue` is, but it sounds pretty scary rather than routine.